sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -58,6 +59,78 @@ def download_gcs_blob(local_path, gcs_path):
     blob = bucket.blob(gcs_blob)
     blob.download_to_filename(local_path)
 
+class DependencyVersion(object):
+  """ DependencyVersion specifies the versions """
+  def __init__(self, version=None, min_version=None, max_version=None):
+    """ if version is specified, no need for min_version or max_version;
+     if both are specified, version is adopted """
+    if version is not None:
+      self._min_version = version
+      self._max_version = version
+    else:
+      self._min_version = min_version
+      self._max_version = max_version
+
+  @property
+  def min_version(self):
+    return self._min_version
+
+  @min_version.setter
+  def min_version(self, min_version):
+    self._min_version = min_version
+
+  def has_min_version(self):
+    return self._min_version != None
+
+  @property
+  def max_version(self):
+    return self._max_version
+
+  @max_version.setter
+  def max_version(self, max_version):
+    self._max_version = max_version
+
+  def has_max_version(self):
+    return self._max_version != None
+
+  def has_versions(self):
+    return (self.has_min_version()) or (self.has_max_version())
+
+
+class DependencyHelper(object):
+  """ DependencyHelper manages software dependency information """
+  def __init__(self):
+    self._PYTHON_PACKAGE = 'PYTHON_PACKAGE'
+    self._dependency = {self._PYTHON_PACKAGE:OrderedDict()}
+
+  @property
+  def python_package(self):
+    return self._dependency[self._PYTHON_PACKAGE]
+
+  def add_python_package(self, name, version=DependencyVersion(), override=True):
+    """ add_single_python_package adds a dependency for the python package
+
+    Args:
+      name: package name
+      version: it could be a specific version(1.10.0), or a range(>=1.0,<=2.0)
+        if not specified, the default is resolved automatically by the pip system.
+      override: whether to override the version if already existing in the dependency.
+    """
+    if name in self.python_package and not override:
+      return
+    self.python_package[name] = version
+
+  def generate_pip_requirements(self, target_file):
```

Comment:
Technically it could be a string. The motivation of generating the requirement file is a cleaner design and it seems to be a standard approach for metaparticle, fair, etc.